app/http.php

```diff
@@ -242,110 +258,139 @@ function logError(Throwable $error, string $action, ?Logger $logger, Utopia\Rout
+            \curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
+            \curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+            \curl_setopt($ch, CURLOPT_HEADER, true);
+            \curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
+            \curl_setopt($ch, CURLOPT_TIMEOUT, \intval(App::getEnv('OPR_PROXY_MAX_TIMEOUT', '900')));
 
-        $headers = \array_merge($request->getHeaders(), [
-            'authorization' => 'Bearer ' . App::getEnv('OPR_PROXY_EXECUTOR_SECRET', '')
-        ]);
+            $curlHeaders = [];
+            foreach ($headers as $header => $value) {
+                $curlHeaders[] = "{$header}: {$value}";
+            }
 
-        // Header used for testing
-        if (App::isDevelopment()) {
-            $headers = \array_merge($headers, [
-                'x-opr-executor-hostname' => $hostname
-            ]);
-        }
+            \curl_setopt($ch, CURLOPT_HTTPHEADER, $curlHeaders);
 
-        $body = $request->getRawPayload();
+            $executorResponse = \curl_exec($ch);
 
-        $ch = \curl_init();
+            $statusCode = \curl_getinfo($ch, CURLINFO_HTTP_CODE);
 
-        \curl_setopt($ch, CURLOPT_URL, $hostname . $request->getURI());
-        \curl_setopt($ch, CURLOPT_CUSTOMREQUEST, $request->getMethod());
-        \curl_setopt($ch, CURLOPT_POSTFIELDS, $body);
-        \curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
-        \curl_setopt($ch, CURLOPT_HEADER, true);
-        \curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
-        \curl_setopt($ch, CURLOPT_TIMEOUT, \intval(App::getEnv('OPR_PROXY_MAX_TIMEOUT', '900')));
+            $error = \curl_error($ch);
 
-        $curlHeaders = [];
-        foreach ($headers as $header => $value) {
-            $curlHeaders[] = "{$header}: {$value}";
-        }
+            $errNo = \curl_errno($ch);
 
-        \curl_setopt($ch, CURLOPT_HTTPHEADER, $curlHeaders);
+            $header_size = curl_getinfo($ch, CURLINFO_HEADER_SIZE);
+            $headers = substr(\strval($executorResponse), 0, $header_size);
+            $body = substr(\strval($executorResponse), $header_size);
 
-        $executorResponse = \curl_exec($ch);
+            \curl_close($ch);
 
-        $statusCode = \curl_getinfo($ch, CURLINFO_HTTP_CODE);
+            if ($errNo !== 0) {
+                throw new Exception('Unexpected curl error between proxy and executor ID ' . $hostname . ' (' . $errNo .  '): ' . $error);
+            }
 
-        $error = \curl_error($ch);
+            $headersArr = [];
+            if (!empty($headers)) {
+                $headers = preg_split("/\r\n|\n|\r/", $headers);
+                if ($headers) {
+                    foreach ($headers as $header) {
+                        if (\str_contains($header, ':')) {
+                            [$key, $value] = \explode(':', $header, 2);
+                            $headersArr[$key] = $value;
+                        }
+                    }
+                }
+            }
```

Comment:
what are we doing here ?

Reply:
to parse headers into key&value pairs. CURL gave it in weird format I think (response body and headers are part of 1 string, and you need to extract it).

Reply:
Updated, used same logic as on executor.